src/lib.rs

```diff
@@ -0,0 +1,437 @@
+// SPDX-FileCopyrightText: 2022 Profian Inc. <opensource@profian.com>
+// SPDX-License-Identifier: AGPL-3.0-only
+
+#![warn(rust_2018_idioms, unused_lifetimes, unused_qualifications, clippy::all)]
+
+#[macro_use]
+extern crate anyhow;
+pub mod kvm;
+
+use attestation::crypto::{CertReqExt, PrivateKeyInfoExt, TbsCertificateExt};
+use attestation::sgx::Sgx;
+use attestation::snp::Snp;
+use kvm::Kvm;
+
+use std::io::BufRead;
+use std::path::Path;
+use std::sync::{Arc, Once};
+use std::time::{Duration, SystemTime};
+
+use anyhow::Context;
+use axum::body::Bytes;
+use axum::extract::{Extension, TypedHeader};
+use axum::headers::ContentType;
+use axum::response::IntoResponse;
+use axum::routing::{get, post};
+use axum::Router;
+use const_oid::db::rfc5280::{
+    ID_CE_BASIC_CONSTRAINTS, ID_CE_EXT_KEY_USAGE, ID_CE_KEY_USAGE, ID_CE_SUBJECT_ALT_NAME,
+    ID_KP_CLIENT_AUTH, ID_KP_SERVER_AUTH,
+};
+use const_oid::db::rfc5912::ID_EXTENSION_REQ;
+use der::asn1::{GeneralizedTime, Ia5StringRef, UIntRef};
+use der::{Decode, Encode, Sequence};
+use hyper::StatusCode;
+use sec1::pkcs8::PrivateKeyInfo;
+use serde::Deserialize;
+use tower_http::trace::{
+    DefaultOnBodyChunk, DefaultOnEos, DefaultOnFailure, DefaultOnRequest, DefaultOnResponse,
+    TraceLayer,
+};
+use tower_http::LatencyUnit;
+use tracing::{debug, Level};
+use x509::attr::Attribute;
+use x509::ext::pkix::name::GeneralName;
+use x509::ext::pkix::{BasicConstraints, ExtendedKeyUsage, KeyUsage, KeyUsages, SubjectAltName};
+use x509::name::RdnSequence;
+use x509::request::{CertReq, ExtensionReq};
+use x509::time::{Time, Validity};
+use x509::{Certificate, TbsCertificate};
+use zeroize::Zeroizing;
+
+pub const PKCS10: &str = "application/pkcs10";
+pub const BUNDLE: &str = "application/vnd.steward.pkcs10-bundle.v1";
+static TRACING: Once = Once::new();
```

Comment:
This is only used in test code, it does not belong in production code

Reply:
Fixed.